react/core/main.tsx

```diff
@@ -133,17 +134,60 @@ const render = (
     </RenderProvider>
   )
 
-  return canUseDOM
-    ? ((disableSSR || created
-        ? renderDOM<HTMLDivElement>(root, elem)
-        : hydrate(root, elem)) as Element)
-    : renderToStringWithData(root).then(({ markup, renderTimeMetric }) => ({
-        markups: getMarkups(name, markup),
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        maxAge: cacheControl!.maxAge,
-        page,
-        renderTimeMetric,
-      }))
+  if (canUseDOM) {
+    const renderFn = disableSSR || created ? renderDOM : hydrate
+
+    renderFn(root, elem)
+
+    return (undefined as unknown) as Element
```

Comment:
Why not `return renderFn(root, elem)`?

Reply:
it still warns because `render` and `hydrate` are typed to also return `void`, but I can just do the type assertion `as Element`

Reply:
oh, I need to convert it to `unknown` as well.